pkg/sql/schemachanger/scplan/internal/opgen/opgen_check_constraint.go

```diff
@@ -19,16 +19,48 @@ func init() {
 	opRegistry.register((*scpb.CheckConstraint)(nil),
 		toPublic(
 			scpb.Status_ABSENT,
+			to(scpb.Status_WRITE_ONLY,
+				emit(func(this *scpb.CheckConstraint) *scop.MakeAbsentCheckConstraintWriteOnly {
+					return &scop.MakeAbsentCheckConstraintWriteOnly{
+						TableID:               this.TableID,
+						ConstraintID:          this.ConstraintID,
+						ColumnIDs:             this.ColumnIDs,
+						Expression:            this.Expression,
+						FromHashShardedColumn: this.FromHashShardedColumn,
+					}
+				}),
+				// TODO (xiang): Add an operation that updates back-references in potentially
+				// affected types and sequences when implementing `add constraint`.
```

Comment:
I believe it simply involves emitting `UpdateTableBackReferencesInTypes` and `UpdateBackReferencesInSequences` operations just like on the dropping path, in which case you might as well do so right now.